### PR TITLE
add offline setting page

### PR DESCRIPTION
**Pull request by NemTechCN**

### CODE REVIEW

**Comment by decentraliser** [comment]
GenerationHash

- should be a required field
- if the current generation hash is not null, the generationHash field should be defaulted with it

**Comment by decentraliser** [comment]
i18n: keys missing in en-EN
- `divisibility`
- `mosaicName`

**Comment by decentraliser** [comment]
the folder name should be `offline-settings`  instead of  `offline_setting`

File: src/views/setting/offline_setting/OfflineSettingTs.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            ticker
+        })
+        this.$Notice.success({
+            title: this.$t(Message.SUCCESS) + ''
+        })
+        this.resetForm()
+    }
+
+    resetForm() {
+        this.formItems = cloneData(formDataConfig.offsetLineForm)
```
Comment (decentraliser) [inline src/views/setting/offline_setting/OfflineSettingTs.ts]: add this for field validation reset after form submit

`this.$nextTick(() => this.$validator.reset())`

File: src/language/en-US.json [hunk 2]
```diff
@@ -749,6 +749,11 @@
     "Debug_console": "Debug console",
     "Data": "Data",
     "divisibility_can_not_more_than_6": "Divisibility can not be more than 6.",
+    "offline_setting": "Offline setting",
```
[the remark this answers is not in the document]
Comment (NemTechCN) [reply src/language/en-US.json]: fixed

File: src/views/setting/offline_setting/OfflineSettingTs.ts (hunk 1 above)
Comment (NemTechCN) [reply src/views/setting/offline_setting/OfflineSettingTs.ts]: fixed

Comment (decentraliser) [reply src/views/setting/offline_setting/OfflineSettingTs.ts]: not sure that it is related... `this.$validator.reset()` is just here to hide the `errorTooltip `when you reset the form

File: src/views/setting/offline-setting/OfflineSettingTs.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        this.resetForm()
+    }
+
+    resetForm() {
+        this.formItems.generationHash = this.generationHash
+        this.formItems.mosaicId = this.networkCurrency.hex
+        this.formItems.divisibility = this.networkCurrency.divisibility
+        this.formItems.mosaicName = this.networkCurrency.name
+        this.formItems.ticker = this.networkCurrency.ticker
+    }
```
Comment (decentraliser) [inline src/views/setting/offline-setting/OfflineSettingTs.ts]: add this as last line
`this.$nextTick(() => this.$validator.reset())`

**Pull request merged**